ecommerce/extensions/payment/views/lumsxpay.py

```diff
@@ -0,0 +1,181 @@
+                "order_id": request.basket.order_number,
+                "user": request.user,
+                "lms_dashboard_url": get_lms_dashboard_url,
+                "is_paid": False,
+                "support_email": request.site.siteconfiguration.payment_support_email
+            }
+
+        return {}
+
+    def get(self, request):
+        configuration_helpers = request.site.siteconfiguration.edly_client_theme_branding_settings
+        url = configuration_helpers.get('LUMSXPAY_VOUCHER_API_URL')
+
+        if request.user.is_anonymous or not (url and request.basket):
+            msg = 'user is anonymous cannot proceed to checkout page so redirecting to login'
+            if not url:
+                msg = 'Site configuration in ecommerce does not include payment API url'
+
+            logger.info(msg)
+            return redirect_to_login(get_lms_dashboard_url)
+
+        if self.is_basket_and_challan_exists(request):
+            context = self.request_already_existing_challan(request)
+            if not context:
+                logger.exception('challan status API not working, no context found')
+                return HttpResponseNotFound()
+
+            return render_to_response('payment/lumsxpay.html', context)
+
+        items = self.extract_items_from_basket(request.basket)
+        payload = {
+            "name": request.user.username,
+            "email": request.user.email,
+            "order_id": request.basket.order_number,
+            "items": items,
+            "due_date": self.get_due_date(configuration_helpers)
+        }
+
+        headers = {
+            "Authorization": configuration_helpers.get('PAYMENT_AUTHORIZATION_KEY'),
+            "Content-Type": "application/json"
+        }
+
+        try:
+            response = requests.post(url, data=json.dumps(payload), headers=headers)
+        except:
+            logger.exception('Challan generation API not working and cannot be reached.')
+            return HttpResponseNotFound()
+
+        if response.status_code == 200:
+            voucher_details = response.json()
+            url_for_online_payment = voucher_details.get('data', {}).get("url_for_online_payment")
+            url_for_download_voucher = voucher_details.get('data', {}).get("url_for_download_voucher")
+
+            context = {
+                'configuration_helpers': configuration_helpers,
+                'url_for_online_payment': url_for_online_payment,
+                'url_for_download_voucher': url_for_download_voucher,
+                'items_list': items,
+                "name": request.user.username,
+                "email": request.user.email,
+                "order_id": request.basket.order_number,
+                "user": request.user,
+                "lms_dashboard_url": get_lms_dashboard_url,
+                "is_paid": False,
+                "support_email": request.site.siteconfiguration.payment_support_email
+            }
+
```

Comment:
I believe this is repeated `request_already_existing_challan`

Reply:
I dont get what you trying to say here?